src/usdc-balance-change/usdc-balance-change.spec.js

```diff
@@ -0,0 +1,185 @@
+  Finding, FindingSeverity, FindingType,
+} = require('forta-agent');
+const { provideInitialize, provideHandleBlock } = require('./usdc-balance-change');
+
+const data = {};
+
+const TEST_ADDRESS = ethers.constants.AddressZero;
+
+// mock the usdc contract
+const mockUsdcContract = {
+  balanceOf: jest.fn(),
+};
+
+describe('USDC balance change', () => {
+  let handleBlock;
+
+  beforeEach(async () => {
+    // set up handler
+    await (provideInitialize(data))();
+
+    handleBlock = provideHandleBlock(data);
+
+    // assign values for testing
+    data.blockWindow = 4;
+    data.pctChangeThreshold = 10;
+
+    // reset the addresses object
+    // use a single address for testing
+    data.addresses = {
+      [TEST_ADDRESS]: {
+        name: 'Test',
+        balanceHistory: [],
+      },
+    };
+
+    mockUsdcContract.balanceOf.mockReset();
+    data.usdcContract = mockUsdcContract;
+  });
+
+  // convenience function to iterate over several blocks to build up balance history
+  // while expecting no findings
+  async function iterateHandlerExpectEmpty(iterations) {
+    for (let i = 0; i < iterations; i++) {
+      // these iterations need to happen synchronously to ensure the ordering is correct
+      // eslint-disable-next-line no-await-in-loop
+      const findings = await handleBlock();
+      expect(findings).toStrictEqual([]);
+    }
+  }
+
+  describe('does not report', () => {
+    it('when the balances do not change', async () => {
+      // simulate a constant balance - 4 USDC
+      mockUsdcContract.balanceOf.mockResolvedValue(ethers.BigNumber.from(4000000));
+
+      // get N balance updates (N=blockWindow)
+      await iterateHandlerExpectEmpty(data.blockWindow);
+
+      // the next block event will give us N+1 balances --> will trigger a balance check
+      const findings = await handleBlock();
+
+      // there should be no findings
+      expect(findings).toStrictEqual([]);
+
+      // do some additional checks on the balance history
+      const { balanceHistory } = data.addresses[TEST_ADDRESS];
+
+      // the current balance should match the last call to balanceOf()
+      expect(balanceHistory.slice(-1)[0]).toStrictEqual(new BigNumber(4000000));
+
+      // there should be 1 less balance recorded for each address at this point
+      // (the oldest was shifted out of the balance array for the next cycle)
+      expect(balanceHistory.length).toStrictEqual(data.blockWindow);
+    });
+
+    it('when all balance changes are below the threshold', async () => {
+      // simulate a changing balance under the threshold
+      // 5.0, 5.1, 5.2, 5.3, 5.4 USDC
+      mockUsdcContract.balanceOf = jest.fn()
+        .mockResolvedValueOnce(ethers.BigNumber.from(5000000))
+        .mockResolvedValueOnce(ethers.BigNumber.from(5100000))
+        .mockResolvedValueOnce(ethers.BigNumber.from(5200000))
+        .mockResolvedValueOnce(ethers.BigNumber.from(5300000))
+        .mockResolvedValueOnce(ethers.BigNumber.from(5400000));
```

Comment:
Minor point: Create a variable to store `5400000` to use here and in the `expect` on line 102.

Reply:
Discussed offline; will leave as-is